Add quarterly return acceptance test

robertparkinson opened the pull request:
https://eaflood.atlassian.net/browse/WATER-4892

This PR updates the return log acceptance tests to include minor changes made to the existing tests and to add in a new test file for the quarterly returns.

robertparkinson committed:
add quarterly return acceptance test

robertparkinson committed:
Merge branch 'main' into return-logs-quarterly-sceanrios

robertparkinson committed:
Merge branch 'main' into return-logs-quarterly-sceanrios

robertparkinson committed:
Merge branch 'main' into return-logs-quarterly-sceanrios

robertparkinson committed:
update quarterly tests

robertparkinson committed:
Merge branch 'main' into return-logs-quarterly-sceanrios

Cruikshanks commented on cypress/fixtures/quarterly-return-logs.json:
Take a look at the changes we made in [Update return log tests after reference change](https://github.com/DEFRA/water-abstraction-acceptance-tests/pull/165).

We were able to strip out anything unrelated, such as charge versions and their associated records. Plus, as only one licence is used in the test, we only set up that licence in the fixture.

You might find it easier to start with one of `return-logs-historic` fixtures and make whatever tweaks you need for quarterly. It might even be that one of those fixtures could be used for this quarterly test as well 🤷

Cruikshanks reviewed (changes requested):
There's a lot of data being set up in the fixture I don't believe we need.

Now would be the time to trim it down me thinks!

Cruikshanks committed:
Update returns tests to match change to badges (#168)

https://eaflood.atlassian.net/browse/WATER-5006

We made a change to **water-abstraction-ui** to [Use new link and tag for returns in search results](https://github.com/DEFRA/water-abstraction-ui/pull/2714).

This meant we made some changes to the badges that the legacy UI displays.

One of the changes was to stop wasting time Title casing the text for the badges as the GOV.UK elements display the text in all uppercase anyway. This broke `cypress/e2e/internal/returns/record-receipt.cy.js`, hence a minor tweak is needed.

The other was to display 'not due yet' instead of 'due' for return logs that are more than 28 days before their due date. This broke `cypress/e2e/external/view-returns.cy.js`.

dependabot[bot] committed:
Bump cypress from 14.5.0 to 14.5.1 (#169)

dependabot[bot] committed:
Bump cypress-mochawesome-reporter from 3.8.2 to 3.8.4 (#170)

robertparkinson committed:
Update acceptance tests after refactor of views (#171)

https://eaflood.atlassian.net/browse/WATER-5139

Now that we have completed the return version set up journey it was time to revisit the views and standardise the code across them. As multiple people were working on the different pages it is always the case that there will be minor variations in how things are implemented.

This PR updates the acceptance tests based on the new updated views.

dependabot[bot] committed:
Bump cypress from 14.5.1 to 14.5.2 (#174)

StuAA78 committed:
Update returns acceptance tests (#173)

https://eaflood.atlassian.net/browse/WATER-5157

We update the `internal/returns/` acceptance tests to reflect recent changes.

StuAA78 committed:
Additional returns test fixes (#175)

After our initial sweep of returns tests, we return and do a more thorough sweep for broken returns tests. This finds a couple more cases to be fixed.

dependabot[bot] committed:
Bump cypress from 14.5.2 to 14.5.3 (#176)

Jozzey committed:
Update packages to latest versions (#178)

Snyk has identified some vulnerabilities in the packages so we are going to update them to the latest versions to resolve this.

The only breaking change in a major update to `cypress-mochawesome-reporter` is that it requires node V22+ which is fine as that's the version we are on.

Jozzey committed:
Switch to using Snyk CLI action in CI (#179)

https://eaflood.atlassian.net/browse/WATER-5191

With the adoption of SSO in the Defra GitHub org, our current [Snyk](https://snyk.io) status checks have stopped working.

The [team issue](https://github.com/DEFRA/water-abstraction-team/issues/144) goes into more detail, but in essence, whilst the GitHub account that **Snyk** uses to connect has SSO enabled, it won't connect to GitHub to update the PR with the test result.

This change performs the same check without **Snyk** having to authenticate to GitHub. Instead, GitHub will authenticate with **Snyk** by using its CLI via a GitHub action.

It does mean **Snyk** will no longer be listed as a separate check; instead, it will just be a new step in our CI. So, the outcome (adding a package with a vulnerability will cause the PR to fail checks) will be the same.

Cruikshanks committed:
Add snyk exception for tmp (#182)

https://github.com/DEFRA/water-abstraction-team/issues/144

**Snyk** has identified that [tmp](https://github.com/raszi/node-tmp) has a [Symlink Attack vulnerability](https://security.snyk.io/vuln/SNYK-JS-TMP-11501554).

It is introduced via [cypress](https://github.com/cypress-io/cypress). Looking at their releases, they are pretty hot on resolving vulnerabilities; they just haven't gotten to this one yet (it's very new).

We have faith they will, so we are happy to add an exception for now.

dependabot[bot] committed:
Bump cypress from 14.5.3 to 14.5.4 (#180)

Cruikshanks committed:
Update billing test due to changed charges (#181)

We know there are uplifts in charges planned for this year, and we have a test that asserts a value is equal to `x`.

Really, it should just be testing a value exists rather than the specific charge because these do change each year. So, this change updates the test to avoid it failing again in future years.

Cruikshanks committed:
Update bill run get() calls post template std work (#183)

https://eaflood.atlassian.net/browse/WATER-5154

We have been working through the views in [water-abstraction-system](https://github.com/DEFRA/water-abstraction-system) to ensure they are consistent. We refer to this as "template standard".

This means some of the current Cypress `cy.get()` calls need to be updated because of these changes, else the tests fail.

robertparkinson committed:
merge with main

robertparkinson committed:
Merge branch 'main' into return-logs-quarterly-sceanrios

robertparkinson committed:
merge issues

robertparkinson committed:
refactor how we build json objects

robertparkinson committed:
Merge branch 'main' into return-logs-quarterly-sceanrios

robertparkinson committed:
fix linting issues

robertparkinson committed:
Merge branch 'main' into return-logs-quarterly-sceanrios

robertparkinson committed:
Merge branch 'main' into return-logs-quarterly-sceanrios

robertparkinson committed:
refactor to pull model builder into funtions

robertparkinson committed:
Merge branch 'main' into return-logs-quarterly-sceanrios

robertparkinson committed:
refactor folder structure and rename files

robertparkinson committed:
fix linter
